src/status_im/ui/screens/wallet/main/views.cljs

```diff
@@ -88,9 +85,11 @@
       [react/text {:style styles/add-asset-text}
        (i18n/label :t/wallet-add-asset)]]]]])
 
-(defn render-asset [{:keys [name symbol icon decimals amount]}]
+(defn render-asset [{:keys [name symbol icon decimals amount] :as asset}]
   (if name ;; If no 'name' then this the dummy value used to render `add-asset`
-    [list/touchable-item #(utils/show-popup "TODO" (str "Details about " symbol " here"))
+    ;; TODO(goranjovic): refactor to remove double dispatch
+    [list/touchable-item  #(do (rf/dispatch [:set :wallet-selected-asset asset])
```

Comment:
When need a better way to pass data to screens..

Reply:
That we definitely do. I have an idea around that, but need to research the navigation support first.

Reply:
Please have a look at the `status-im.chat.event` namespace and how the navigation to chat screen combined with preselecting data works there, there is really no need for this double dispatch. 
We should avoid using the `:set`/`:set-in` event like a plaque (same goes for equivalent subscriptions, hopefully, it will be removed soon) and really not introduce any new code doing this.